### PR TITLE
[#1051-3] Feature/Haalcentraal BRP data view

### PR DESCRIPTION
**Pull request by vaszig**

### CODE REVIEW

File: src/open_inwoner/accounts/views/profile.py [hunk 1]
```diff
@@ -149,6 +152,62 @@ def form_valid(self, form):
...
+            if brp_version == "2.0"
+            else "geslachtsaanduiding",
+            "street": "verblijfplaats.straat",
+            "house_number": "verblijfplaats.huisnummer",
+            "postcode": "verblijfplaats.postcode",
+            "place": "verblijfplaats.woonplaats",
+            "land": "verblijfplaats.land.omschrijving",
+        }
+
+        fetched_data = fetch_brb_data(user, brp_version)
```
Comment (alextreme) [inline src/open_inwoner/accounts/views/profile.py:193]: fetch_brp_data, not brb :)

File: src/open_inwoner/haalcentraal/signals.py [hunk 2]
```diff
@@ -1,79 +1,19 @@
...
 
 from open_inwoner.accounts.choices import LoginTypeChoices
 from open_inwoner.accounts.models import User
-from open_inwoner.haalcentraal.models import HaalCentraalConfig
 from open_inwoner.utils.logentry import system_action
 
-logger = logging.getLogger(__name__)
-
-
-def fetch_data(instance, brp_version):
```
Comment (alextreme) [inline src/open_inwoner/haalcentraal/signals.py:21]: Good to move this out of signals.py

**Review by alextreme — CHANGES_REQUESTED** [review]
Glad to see this taking shape, nice work!

In your fetch_brp_data utils function please also log to timelinelogger (utils.user_action) that a User (object) requests the BRP data. This to provide visibility on when external calls have been performed.

File: src/open_inwoner/accounts/views/profile.py [hunk 3]
```diff
@@ -149,6 +153,64 @@ def form_valid(self, form):
...
+            "land": "verblijfplaats.land.omschrijving",
+        }
+
+        self.log_user_action(user, _("user requests for brp data"))
+
+        fetched_data = fetch_brp_data(user, brp_version)
+
+        # we have a different response depending on brp version
+        if brp_version == "2.0" and fetched_data.get("personen"):
+            fetched_data = glom(fetched_data, "personen")[0]
```
Comment (Bartvaderkin) [inline src/open_inwoner/accounts/views/profile.py:200]: You don't need to glom if it is only one level deep `fetched_data["personen"][0]`

File: src/open_inwoner/haalcentraal/utils.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+def fetch_brp_data(instance, brp_version):
+    config = HaalCentraalConfig.get_solo()
+
+    if not config.service:
+        logger.warning("no service defined for Haal Centraal")
+        return {}
+
+    client = config.service.build_client()
+    logger.warning(brp_version)
```
Comment (Bartvaderkin) [inline src/open_inwoner/haalcentraal/utils.py:22]: This warning feels a bit lost.

File: src/open_inwoner/accounts/views/profile.py [hunk 5]
```diff
@@ -149,6 +153,64 @@ def form_valid(self, form):
...
+            "first_name": "naam.voornamen",
+            "initials": "naam.voorletters",
+            "last_name": "naam.geslachtsnaam",
+            "prefix": "naam.voorvoegsel",
+            "birthday": "geboorte.datum.datum",
+            "birthday_place": "geboorte.plaats.omschrijving",
+            "birthday_country": "geboorte.land.omschrijving",
+            "gender": "geslachtsaanduiding.omschrijving"
+            if brp_version == "2.0"
+            else "geslachtsaanduiding",
```
Comment (Bartvaderkin) [inline src/open_inwoner/accounts/views/profile.py:186]: Please put parenthesis around inline if/else so it black's for better visibility.

File: src/open_inwoner/accounts/views/profile.py [hunk 6]
```diff
@@ -149,6 +153,64 @@ def form_valid(self, form):
...
+            fetched_data = glom(fetched_data, "personen")[0]
+
+        if not fetched_data:
+            return {}
+
+        for field in my_data:
+            try:
+                my_data[field] = glom(fetched_data, my_data[field])
+            except PathAccessError:
+                my_data[field] = None
```
Comment (Bartvaderkin) [inline src/open_inwoner/accounts/views/profile.py:209]: `glom()` has a `default` parameter so you don't need the try/except.

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]

**Review by Bartvaderkin — APPROVED** [review]

**Review by alextreme — APPROVED** [review]

**Pull request merged**